src/Interpreters/ExpressionActions.cpp

```diff
@@ -351,11 +463,24 @@ static void executeAction(const ExpressionActions::Action & action, ExecutionCon
                     arguments[i] = columns[action.arguments[i].pos];
             }
 
-            ProfileEvents::increment(ProfileEvents::FunctionExecute);
-            if (action.node->is_function_compiled)
-                ProfileEvents::increment(ProfileEvents::CompiledFunctionExecute);
+            /// Use lazy execution when:
+            ///  - It's force enabled.
+            ///  - It's is enabled and function is suitable for lazy execution or it has lazy executed arguments.
+            if (action.node->lazy_execution == ActionsDAG::LazyExecution::FORCE_ENABLED
+                || (action.node->lazy_execution == ActionsDAG::LazyExecution::ENABLED
+                    /*&& (action.node->function_base->isSuitableForShortCircuitArgumentsExecution(arguments)
+                        || checkShirtCircuitArguments(arguments) != -1)*/))
```

Comment:
I commented it, because I want to see what if we enable short-circuit for all functions (it can help find some bugs)